### PR TITLE
Fix ifdef bug from previous PR

### LINKED ISSUE #855: ifdef statements in pr 851 are not evaluated as expected
**Describe the bug**
The change to allow charn to be reset for st3 or st4 does not produce the expected code.

**To Reproduce**

Found by testing the UWM model after updating to develop, which does not reproduce. 

Can also be found by using cpp directly on code:

``cpp -DW3_IC0 -DW3_ST4 w3updtmd.F90``

```
            VA(:,JSEA) = 0.
#ifdef W3_ST3 || defined(W3_ST4)
            CHARN(JSEA) = AALPHA
#else
            CHARN(JSEA) = 0.
#endif
```

produces 

```
          VA(:,JSEA) = 0.



          CHARN(JSEA) = 0.
```

**Expected behavior**

When IC0 and ST4 are active, cpp should produce
```
          VA(:,JSEA) = 0.



          CHARN(JSEA) = AALPHA
```

The expected behaviour can be produced with:

#if defined W3_ST3 || defined(W3_ST4)

### PR DESCRIPTION
**Pull request by JessicaMeixner-NOAA**
# Pull Request Summary
A fix for using two switches incorrectly 

## Description
This is a follow up fix to PR #851 to fix bug ifdef definition for two switches

### Issue(s) addressed
- fixes #855

### Check list  

- [x] Branch is up to date with the authoritative repository (NOAA-EMC) develop branch. 
- [x] Checked the [checklist for a developer submitting to develop](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-a-developer-submitting-to-develop). 
- [ ] If a version number update is required, checked the [updating version number](https://github.com/NOAA-EMC/WW3/wiki/Code-Management#checklist-for-updating-version-number) checklist. 
- [ ] If a new feature was added, a regression test for testing the new feature is added. 
 
### Testing

* How were these changes tested? Matrix regtests + merging this branch to def/ufs-weather-model and running coupled control and debug tests 
* Are the changes covered by regression tests? Yes
* Have the matrix regression tests been run (if yes, please note HPC and compiler)? hera intel 2018 (after bug fix intel 2022 )
* Please indicate the expected changes in the regression test output, (Note the [list](https://github.com/NOAA-EMC/WW3/wiki/How-to-use-matrix.comp-to-compare-regtests-with-develop#4-look-at-results) of known non-identical tests.)  Nothing beyond the expected tests. 

After fixing an additional bug, we do again have difference which mimic those in PR#851, which also had a change in ufs_1.2.  Further testing with PR #851 showed that this was simply in the charn output.  These diffs are anticipated to as well be in the charn output however the run with additional output was not made (can be done if needed). 

* Please provide the summary output of matrix.comp (_matrix.Diff.txt_, _matrixCompFull.txt_ and _matrixCompSummary.txt_):

```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR2_UNO_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (20 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (16 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (12 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
ww3_ta1/./work_UPD0F_U                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.17/./work_ma                     (1 files differ)
ww3_tp2.17/./work_a                     (1 files differ)
ww3_tp2.17/./work_mc1                     (1 files differ)
ww3_tp2.17/./work_mb                     (1 files differ)
ww3_tp2.17/./work_mc                     (1 files differ)
ww3_tp2.17/./work_ma1                     (1 files differ)
ww3_tp2.17/./work_c                     (1 files differ)
ww3_tp2.17/./work_b                     (1 files differ)
ww3_tp2.6/./work_ST0                     (1 files differ)
ww3_tp2.6/./work_ST4                     (1 files differ)
ww3_tp2.6/./work_pdlib                     (1 files differ)
ww3_ufs1.3/./work_a                     (1 files differ)
```

[matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/10068551/matrixCompFull.txt)
[matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/10068552/matrixCompSummary.txt)
[matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/10068553/matrixDiff.txt)

Update after additional bug fix and using newest intel: 
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR1_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (9 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
ww3_ta1/./work_UPD0F_U                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (6 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.17/./work_a                     (1 files differ)
ww3_tp2.17/./work_c                     (1 files differ)
ww3_tp2.17/./work_b                     (1 files differ)
ww3_tp2.6/./work_ST0                     (1 files differ)
ww3_tp2.6/./work_ST4                     (1 files differ)
ww3_tp2.6/./work_pdlib                     (1 files differ)
ww3_ts4/./work_ug_MPI                     (1 files differ)
ww3_ufs1.2/./work_a                     (2 files differ)
ww3_ufs1.2/./work_b                     (2 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
```

### CODE REVIEW

File: model/src/w3updtmd.F90 [hunk 1]
```diff
@@ -1911,7 +1911,7 @@ SUBROUTINE W3UICE ( VA )
                ICEI(IX,IY), 'ICE (NEW)'
 #endif
           VA(:,JSEA) = 0.
-#ifdef W3_ST3 || defined(W3_ST4)
+#if defined W3_ST3 || defined(W3_ST4)
```
Comment (DeniseWorthen) [inline model/src/w3updtmd.F90:1914]: There is a third location that also needs fixing at LN 1950.

**Comment by JessicaMeixner-NOAA** [comment]
```
**********************************************************************
********************* non-identical cases ****************************
**********************************************************************
mww3_test_03/./work_PR1_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)
mww3_test_03/./work_PR3_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_e                     (1 files differ)
mww3_test_03/./work_PR2_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR1_MPI_d2                     (9 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)
mww3_test_03/./work_PR3_UNO_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR2_UQ_MPI_d2                     (16 files differ)
mww3_test_03/./work_PR3_UQ_MPI_e                     (1 files differ)
mww3_test_03/./work_PR3_UQ_MPI_d2                     (15 files differ)
ww3_ta1/./work_UPD0F_U                     (0 files differ)
ww3_tp2.10/./work_MPI_OMPH                     (6 files differ)
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)
ww3_tp2.17/./work_a                     (1 files differ)
ww3_tp2.17/./work_c                     (1 files differ)
ww3_tp2.17/./work_b                     (1 files differ)
ww3_tp2.6/./work_ST0                     (1 files differ)
ww3_tp2.6/./work_ST4                     (1 files differ)
ww3_tp2.6/./work_pdlib                     (1 files differ)
ww3_ts4/./work_ug_MPI                     (1 files differ)
ww3_ufs1.2/./work_a                     (2 files differ)
ww3_ufs1.2/./work_b                     (2 files differ)
ww3_ufs1.3/./work_a                     (3 files differ)
```

After most recent fix, these are the differences, the ufs1.2 has charn as an output, and based on prior testing this is the only value that actually changes, but we see it in the binary output.  The others are known b4b tests + unstructured grid mod_defs.  The mww3_test_03/./work_*_e tests were also non-b4b with the newest version of intel from the last PR.

**Review by DeniseWorthen — APPROVED** [review]
I tested with the UFS RTs which use waves and everything passed.

File: model/src/w3updtmd.F90 (hunk 1 above)
Comment (MatthewMasarik-NOAA) [reply model/src/w3updtmd.F90:1914]: @DeniseWorthen thank you for testing with UFS and confirming it passes.  Regarding the third location, @JessicaMeixner-NOAA made an update earlier today with that last case fixed.  Thank you for catching that.

**Review by MatthewMasarik-NOAA — APPROVED** [review]
These updated `#if`'s were checked by running `cpp` on `w3updtmd.F90`, as described in #855 (thanks for the tip @DeniseWorthen!).  In each case `CHARN(JSEA) = AALPHA`, as intended.

Summary matrix.comp for tests run on `hera.intel`:
```bash
**********************************************************************       
********************* non-identical cases ****************************       
**********************************************************************       
mww3_test_03/./work_PR1_MPI_e                     (1 files differ)           
mww3_test_03/./work_PR3_UQ_MPI_e_c                     (1 files differ)      
mww3_test_03/./work_PR2_UNO_MPI_d2                     (16 files differ)     
mww3_test_03/./work_PR1_MPI_d2                     (8 files differ)          
mww3_test_03/./work_PR3_UNO_MPI_d2_c                     (10 files differ)   
mww3_test_03/./work_PR3_UQ_MPI_d2_c                     (15 files differ)    
mww3_test_03/./work_PR3_UNO_MPI_d2                     (16 files differ)     
mww3_test_03/./work_PR2_UQ_MPI_d2                     (16 files differ)      
mww3_test_03/./work_PR3_UQ_MPI_d2                     (16 files differ)      
ww3_ta1/./work_UPD0F_U                     (0 files differ)                  
ww3_tp2.10/./work_MPI_OMPH                     (7 files differ)              
ww3_tp2.16/./work_MPI_OMPH                     (4 files differ)              
ww3_tp2.6/./work_ST0                     (1 files differ)                    
ww3_tp2.6/./work_ST4                     (1 files differ)                    
ww3_tp2.6/./work_pdlib                     (1 files differ)                  
ww3_ts4/./work_ug_MPI                     (1 files differ)                   
ww3_ufs1.2/./work_a                     (2 files differ)                     
ww3_ufs1.2/./work_b                     (2 files differ)                     
ww3_ufs1.3/./work_a                     (3 files differ)                     
**********************************************************************       
************************ identical cases *****************************       
**********************************************************************
```
* [matrixCompFull.txt](https://github.com/NOAA-EMC/WW3/files/10071800/matrixCompFull.txt)
* [matrixCompSummary.txt](https://github.com/NOAA-EMC/WW3/files/10071801/matrixCompSummary.txt)
* [matrixDiff.txt](https://github.com/NOAA-EMC/WW3/files/10071802/matrixDiff.txt)

The only _unexpected_ differences seen in the Summary output above are in `ww3_ufs1.2`, which was addressed above as being the value of `CHARN` output in those tests.

Based on these regression test results as well as the code review I approve this PR.  Thank you for this update, @JessicaMeixner-NOAA.

**Pull request merged**